src/drivers/ll40ls/LidarLitePWM.cpp

```diff
@@ -196,6 +200,27 @@ int LidarLitePWM::measure()
 		return reset_sensor();
 	}
 
+	uint32_t pulse_interval = _range.timestamp - _lastTimeStamp;
+	_lastTimeStamp = _range.timestamp;
+
+	// Normal reporting interval for LidarLite in PWM mode is 50msec
+	if (pulse_interval > 60 * 1000UL) { // TODO : Should we increase this to *not* miss longer range meas?
+		// missed pulse, frequently associated with invalid pulsewidth measurements
+		_pulseCount = 0;
+
+	} else {
```

Comment:
It might be better to handle this using fault detection instead of waiting for 4 valid reads.